src/schema/v2/HeroUnit/updateHeroUnitMutation.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  GraphQLInt,
+  GraphQLObjectType,
+  GraphQLString,
+  GraphQLUnionType,
+} from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { GraphQLNonNull } from "graphql"
+import { ResolverContext } from "types/graphql"
+import { HeroUnitType } from "./HeroUnitType"
+import {
+  formatGravityError,
+  GravityMutationErrorType,
+} from "lib/gravityErrorHandler"
+
+interface Input {
+  body: string
+  credit?: string
+  endAt?: string
+  id: string
+  imageUrl?: string
+  label?: string
+  linkText: string
+  linkUrl: string
+  position?: number
+  startAt?: string
+  title: string
+}
+
+const SuccessType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateHeroUnitSuccess",
+  isTypeOf: (data) => data.id,
+  fields: () => ({
+    heroUnit: {
+      type: HeroUnitType,
+      resolve: (hero_unit) => hero_unit,
+    },
+  }),
+})
+
+const FailureType = new GraphQLObjectType<any, ResolverContext>({
+  name: "updateHeroUnitFailure",
+  isTypeOf: (data) => data._type === "GravityMutationError",
+  fields: () => ({
+    mutationError: {
+      type: GravityMutationErrorType,
+      resolve: (err) => err,
+    },
+  }),
+})
+
+const ResponseOrErrorType = new GraphQLUnionType({
+  name: "updateHeroUnitResponseOrError",
+  types: [SuccessType, FailureType],
+})
+
+export const updateHeroUnitMutation = mutationWithClientMutationId<
+  Input,
+  any | null,
+  ResolverContext
+>({
+  name: "UpdateHeroUnitMutation",
+  description: "updates a hero unit.",
+  inputFields: {
+    id: { type: new GraphQLNonNull(GraphQLString) },
+    body: { type: new GraphQLNonNull(GraphQLString) },
```

Comment:
Depending, you may or may not want these to all be required for an 'update'. Do you want to support a client that wants to update _just_ the `label`, for instance, and not send the other fields over?

Reply:
So far the only client here is Forque which sends all so I just did what was easiest. 🤷